### PR TITLE
added more test cases and grouped them into multiple tests

### PR DESCRIPTION
**Pull request by zehua**
test test test

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -6,13 +6,49 @@
...
   assert.strictEqual(leftPad('foobar', 6), 'foobar');
+  assert.strictEqual(leftPad('foobar', 5), 'foobar');
+  assert.strictEqual(leftPad('foobar', -1), 'foobar');
+  assert.strictEqual(leftPad('foobar', 6, '1'), 'foobar');
+  assert.strictEqual(leftPad('foobar', 5, '1'), 'foobar');
+  assert.strictEqual(leftPad('foobar', -1, '1'), 'foobar');
+  assert.strictEqual(leftPad('foobar', 8, ''), '  foobar');
+  assert.strictEqual(leftPad('foobar', 8, false), '  foobar', 'false default to space');
+  //assert.strictEqual(leftPad('foobar', 8, []), '  foobar', '[] default to space');
+  //assert.strictEqual(leftPad('foobar', 8, {}), '  foobar', '{} default to space');
```
Comment (zehua) [inline test.js]: These two currently fail. Not sure if we should make them pass by updating the code.

Comment (stevemao) [reply test.js]:
``` js
String([])
""

String({})
"[object Object]"
```

The second argument should be always a string, even though we are not forcing it here (we using `String()` to convert anything).

Comment (zehua) [reply test.js]: yeah, I know what the current behaviour is. Passing `[]` will result in nothing being padded, and `{}` with `[object Object]` being padded. They don't look right, but I am just not sure what the desired behaviour should be.

I prefer to handle them in a separate PR. We could enable these two tests at that time.

Comment (stevemao) [reply test.js]: @zehua I understand. I don't see a module should handle everything. users should understand the contract. If not, and if it's a problem, they might already have opened an issue with real-life scenarios :) But yeah it's nothing right or wrong here, just a different paradigm I believe 😄

**Comment by zehua** [comment]
do you want me to remove these two test cases, or leave them commented here?

**Review by stevemao — CHANGES_REQUESTED** [review]
remove them. I don't think they are valid use-cases. Unless you have strong opinions?

**Comment by zehua** [comment]
Removed the commented tests. @stevemao

**Pull request merged**